### DISCUSSION TITLE
Mesh refinement based on a material property

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#24959] [Q&A Meshing]
Hi,

We defined a material property for the stored energy, which is non-zero on the dislocations present in the sub-grain structure. For example, the 2D structure looks as follows in the image shown below:
![400_by_400_rayleigh](https://github.com/idaholab/moose/assets/14978999/af58e0ae-9d17-4965-8eb9-d19b801cbc3a)

In the above figure, the horizontal lines are GNBs, and the vertical lines are IDBs; in other words, they are dislocation boundaries. Since this structure is instantiated at the beginning of the simulation, I would like to refine my uniform mesh on the areas where the dislocation boundaries exist. Any suggestions?

Thanks!

**Comment by GiudGiud** [comment]
is this information (where you want to refine) only stored in the material property?

if so, move it to an auxiliary variable, use that as an indicator then use initial adaptivity
https://mooseframework.inl.gov/source/actions/AdaptivityAction.html

**Reply by rayaprolu143** [reply]
Thank you! I will implement this idea of yours.
